server/middleware/memcached/plugin.js

```diff
@@ -0,0 +1,41 @@
+const Memcached = require('memcached');
+
+require('dotenv').config();
+
+class MemoryCache {
+  constructor() {
+    const host = process.env.CACHE_HOST || '127.0.0.1';
+
+    this.cache = new Memcached(`${host}:11211`);
+  }
+
+  retrieve = async (key) => {
+    const cache = this.cache;
+
+    return new Promise((resolve, reject) => {
+      cache.get(key, (err, data) => {
+        if (err) {
+          reject(err);
+        } else {
+          resolve(data);
+        }
+      });
+    });
+  };
+
+  store = async (key, value, lifetime) => {
+    const cache = this.cache;
+
+    return new Promise((resolve, reject) => {
+      cache.set(key, value, lifetime * 60, (err) => {
+        if (err) {
+          reject(err);
+        } else {
+          resolve(true);
+        }
+      });
+    });
+  };
+}
+
+module.exports = MemoryCache;
```

Comment:
Looks good.
I'm not as familiar with memcached, I assume that you'll be able to use as written for when a post request is sent to update values?

Reply:
Yes, if the key (currently set as the url) is the same, it will replace the contents.